File: java/src/Solution_Sword_10_1.java
public class Solution_Sword_10_1 {
    public int fib(int n) {
        int a = 0, b = 1, sum;
        for (int i = 0; i < n; ++i) {
            sum = (a + b) % 1000000007;
            a = b;
            b = sum;
        }
        return a;
    }
}

File: python/Solution_Sword_10_1.py
from functools import lru_cache


class Solution:
    @lru_cache(None)
    def fib(self, n):
        if n < 2:
            return n
        return (self.fib(n-1) + self.fib(n-2)) % 1000000007


if __name__ == '__main__':
    print(Solution().fib(1))